src/main/java/sirius/kernel/commons/Files.java

```diff
@@ -146,6 +155,38 @@ public static String getFilenameWithoutExtension(@Nullable String path) {
         return Strings.splitAtLast(getFilenameAndExtension(path), ".").getFirst();
     }
 
+    /**
+     * Determines if the given path is hidden.
+     * <p>
+     * A path is considered hidden if it starts with a dot.
+     *
+     * @param path the path to check
+     * @return <tt>true</tt> if the path is hidden, <tt>false</tt> otherwise
+     */
+    public static boolean isConsideredHidden(@Nullable String path) {
+        try {
+            return getFilenameAndExtension(path).startsWith(".");
+        } catch (NullPointerException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the given path is a metadata file.
+     * <p>
+     * A metadata file is a file which is not part of the actual content but rather contains metadata or is used by
+     * the operating system or other tools.
+     *
+     * @param path the path to check
+     * @return <tt>true</tt> if the path is a metadata file that is listed in the METADATA_FILES list, <tt>false</tt> otherwise
+     */
+    public static boolean isConsideredMetadata(@Nullable String path) {
+        if (Strings.isEmpty(path)) {
+            return false;
+        }
+        return METADATA_FILES.stream().anyMatch(path::startsWith);
```

Comment:
Required: also apply here the same logic to obtain the last portion of the path.
The metadata folder or file could be in a lower level, like `folder1/folder2/__MACOSX`